File: Source/Camping Mod 2.0/common/rikmuld/camping/inventory/slot/SlotItem.java
package rikmuld.camping.inventory.slot;

import net.minecraft.inventory.IInventory;
import net.minecraft.inventory.Slot;
import net.minecraft.item.ItemStack;

public class SlotItem extends Slot {

	public SlotItem(IInventory inv, int id, int x, int y)
	{
		super(inv, id, x, y);
	}

	@Override
	public boolean isItemValid(ItemStack is)
	{
		return is.itemID > 4096? true:is.itemID < 256? false:is.itemID < 421;
	}
}

File: scala/com/rikmuld/camping/inventory/SlotItem.scala
package com.rikmuld.camping.inventory

import com.rikmuld.corerm.gui.slots.SlotOnly
import net.minecraft.inventory.{IInventory, Slot}
import net.minecraft.item.Item

class SlotItem(inv: IInventory, index: Int, x: Int, y: Int, item: Item) extends Slot(inv, index, x, y) with SlotOnly {
  override def getAllowedItems: Vector[Item] =
    Vector(item)
}